backend_jobs/ingestion_pipeline/unit_tests/firestore_database_test.py

```diff
@@ -0,0 +1,108 @@
+        expects to receive a map object with all the geohashes.
+
+        Given:
+            latitude=37.783371, longitude=-122.439687
+
+        Expects:
+            {...'hash10': '9q8yvy60hd', ...}
+        """
+        expected_hashes_map = '9q8yvy60hd'
+        hashes_map = firestore_database._get_geo_hashes_map(37.783371, -122.439687)
+        self.assertEqual(expected_hashes_map, hashes_map['hash10'])
+ 
+    def test_get_geo_hashes_map2(self):
+        """ This test, given a latitude and longitude,
+        expects to receive a map object with all the geohashes.
+
+        Given:
+            latitude=0.0, longitude=0.0
+
+        Expects:
+            {...'hash10': '7zzzzzzzzz', ...}
+        """
+        expected_hashes_map = '7zzzzzzzzz'
+        hashes_map = firestore_database._get_geo_hashes_map(0.0, 0.0)
+        self.assertEqual(expected_hashes_map, hashes_map['hash10'])
+ 
+    def test_get_geo_hashes_map3(self):
+        """ This test, given a latitude and longitude,
+        expects to receive a map object with all the geohashes.
+
+        Given:
+            latitude=-90, longitude=180
+
+        Expects:
+            {...'hash10': 'pbpbpbpbpb', ...}
+        """
+        expected_hashes_map = 'pbpbpbpbpb'
+        hashes_map = firestore_database._get_geo_hashes_map(-90, 180)
+        self.assertEqual(expected_hashes_map, hashes_map['hash10'])
+    
+    def test_get_date_fields(self):
+        """ This test, given a datetime object,
+        expects to receive a map object with the date fields parsed.
+        """
+        expected_date = {'year': 2019, 'month': 4, 'day': 13}
+        date = firestore_database._get_date_fields(datetime.date(2019, 4, 13))
+        self.assertEqual(expected_date, date)
+
+    def test_get_max_visibility_visible(self):
+        """ This test, given two visibilities, expects to receive the max visibility.
+
+        Given:
+            first_visibility=VISIBLE, second_visibility=INVISIBLE
+
+        Expects:
+            VISIBLE .
+        """
+        expected_visibility = data_types.VisibilityType.VISIBLE
+        visibility = firestore_database._get_max_visibility(
+            data_types.VisibilityType.VISIBLE,
+            data_types.VisibilityType.INVISIBLE)
+        self.assertEqual(expected_visibility, visibility)
+
+    def test_get_max_visibility_invisible(self):
+        """ This test, given two visibilities, expects to receive the max visibility.
+        
+        Given:
+            first_visibility=INVISIBLE, second_visibility=INVISIBLE
+
+        Expects:
+            INVISIBLE .
+        """
+        expected_visibility = data_types.VisibilityType.INVISIBLE
+        visibility = firestore_database._get_max_visibility(
+            data_types.VisibilityType.INVISIBLE,
+            data_types.VisibilityType.INVISIBLE)
+        self.assertEqual(expected_visibility, visibility)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
this is very verbose for a very simple test.
Maybe merge these two functions and put both tests in the same function.

Reply:
it might be worth checking out parametrized tests, see:
https://docs.pytest.org/en/stable/example/parametrize.html

Reply:
merge these two functions using parametrized tests.